src/sapling/sapling_operation.h

```diff
@@ -41,6 +41,8 @@ class SaplingOperation {
     explicit SaplingOperation(const Consensus::Params& consensusParams, int chainHeight) : txBuilder(consensusParams, chainHeight) {};
     explicit SaplingOperation(TransactionBuilder& _builder) : txBuilder(_builder) {};
 
+    ~SaplingOperation() { if (managetKeyChange) delete tkeyChange; }
```

Comment:
(c255cded228a1500cafd75171779d21ac776405a)
Can't we just check ` if (tkeyChange)` and remove `managetKeyChange` entirely?

Reply:
yeah, this is a remanent of a previous implementation. We actually can remove the if as well. Deleting a null pointer has no effect.